### DISCUSSION TITLE
Stuck on Loading Screen

### DISCUSSION THREAD

**Discussion by ddtjco** [Kanaries/pygwalker#551] [Q&A]
I recently enabled pygwalker on a streamlit application.  We are hosting the streamlit application locally on a Windows server using a custom url.  If I hit the streamlit application from the localhost url, everything with pygwalker works as expected.  But I access the application using our custom url, the pygwalker screen stays stuck on the "Loading Graphic-Walker UI" screen.  I notice in the inspector tools that the resource (_stcore/_pygwalker/comm/...) returns a 404 error when accessing from our url, but not from the localhost.

**Comment by longxiaofei** [comment]
Hi @ddtjco , thanks for your feedback.

There are many reasons for this, I need some details.

"custom url" means you deployed the streamlit app on your server? If so

1. There are any additional configuration options passed in when deploying streamlit apps?
2. Have you used other reverse proxy tools (such as nginx) to forward streamlit server?
3. I want to know what is the path of streamlit's own communication websocket?
![image](https://github.com/Kanaries/pygwalker/assets/28337703/deb44876-ccd0-4db8-83f2-632ce8dd8a71)

**Comment by ddtjco** [comment]
Thanks for following up.  Turns out we had an error with our url rewrite rules that was causing the issues.  Once we fixed those, it began to work.

**Reply by datndptit** [reply]
Hi, I have the same error as you. How did you fix the error? Can you tell me?

**Comment by datndptit** [comment]
I have solved the problem
I add this in in my /etc/nginx/sites-avaible/mysite
location /_stcore/_pygwalker {
            proxy_pass http://localhost:8501/_stcore/_pygwalker;
            proxy_http_version 1.1;
            proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
            proxy_set_header Host $host;
            proxy_set_header Upgrade $http_upgrade;
            proxy_set_header Connection "upgrade";
            proxy_set_header X-Real-IP $remote_addr;
            proxy_set_header X-Forwarded-Proto $scheme;
 }

**Comment by RowenTey** [comment]
If I deployed Streamlit in Kubernetes and used an Nginx ingress for reverse proxy, how should I approach this issue?

Browser console image below for reference: 
![image](https://github.com/user-attachments/assets/2c7d4635-95fd-4b0a-a50a-8b2587c6a19b)

WebSocket network console image for reference:
![image](https://github.com/user-attachments/assets/1cc0ca9a-499d-4e26-95e0-f163bbefaf73)

**Reply by longxiaofei** [reply]
What is the url of the current page?

**Reply by RowenTey** [reply]
It's `https://fyp.rowentey.xyz/explore`.

**Reply by RowenTey** [reply]
I'm getting an `"Unknown gid: 000624bc402cd41c3nI1FNYcMHzpKTa5"`, any ideas why?
